src/middlewared/middlewared/plugins/interface/link_address.py

```diff
@@ -80,48 +78,54 @@ async def setup(middleware):
         )
 
         # Migrate BSD network interfaces to Linux
-        if osc.IS_LINUX:
-            for db_interface in db_interfaces:
-                if m := RE_FREEBSD_BRIDGE.match(db_interface["interface"]):
-                    name = f"br{m.group(1)}"
-                    await rename_interface(middleware, db_interface, name)
-                    db_interface["interface"] = name
-
-                if m := RE_FREEBSD_LAGG.match(db_interface["interface"]):
-                    name = f"bond{m.group(1)}"
-                    await rename_interface(middleware, db_interface, name)
-                    db_interface["interface"] = name
-
-                if db_interface["link_address"] is not None:
-                    if real_interfaces.by_name.get(db_interface["interface"]) is not None:
-                        # There already is an interface that matches DB cached one, doing nothing
-                        continue
-
-                    real_interface_by_link_address = real_interfaces.by_link_address.get(db_interface["link_address"])
-                    if real_interface_by_link_address is None:
+        for db_interface in db_interfaces:
+            if db_interface["interface"].startswith(("vlan", "br", "bond")):
+                # vlan interfaces dont need to be migrated since they
+                # share the same name between CORE and SCALE
+                # "br" and "bond" interface names are already converted
+                # to scale
+                continue
+
+            if m := RE_FREEBSD_BRIDGE.match(db_interface["interface"]):
+                name = f"br{m.group(1)}"
```

Comment:
Would this condition ever be true ?

Reply:
Why not? Didn't we have `bridgeN` interfaces in the database in CORE?